src/components/views/Order/Order.js

```diff
@@ -7,83 +7,78 @@ import clsx from 'clsx';
 import { connect } from 'react-redux';
 import { getCartItems, cleanCartItems } from '../../../redux/cartRedux.js';
 import { getPersonalData, addOrderInAPI, cleanOrderForm } from '../../../redux/orderRedux.js';
+import { getOrderValidation } from '../../../redux/validationRedux';
 
 import styles from './Order.module.scss';
+import { getCurrentDate } from '../../../utils';
+import { inputFields } from './PersonalDataForm/config';
 import { OrderListItem } from '../../features/OrderListItem/OrderListItem';
-import { PersonalDataForm } from '../../features/PersonalDataForm/PersonalDataForm';
+import { PersonalDataForm } from './PersonalDataForm/PersonalDataForm';
 import { OrderSummary } from '../../features/OrderSummary/OrderSummary';
 import { Button } from '../../common/Button/Button';
 
 import Container from 'react-bootstrap/Container';
 import Col from 'react-bootstrap/Col';
 import Modal from 'react-bootstrap/Modal';
 
-const Component = ({ className, cartItems, addOrderInAPI, personalData, cleanCartItems, cleanOrderForm }) => {
+const Component = ({ className, ...props }) => {
+  const {
+    cartItems,
+    addOrderInAPI,
+    personalData,
+    cleanCartItems,
+    cleanOrderForm,
+    orderValidation,
+  } = props;
+
+  const requiredFields = Object.keys(inputFields)
+    .map(item => inputFields[item])
+    .filter(item => item.validationRules.required)
+    .map(item => item.name);
```

Comment:
Trzy pętle. Niby nic, ale można by się zastanowić nad zmianą na dwie, może nawet jedną, nawet jeśli oznaczałoby to wykorzystanie klasycznej pętli for(). Czytelność może potencjalnie troszkę spaść, ale kod będzie działać trochę szybciej. Dla tego formularza pewnie nie będzie to specjalnie istotne, ale może być w innych sytuacjach.